test/partitioning.jl

```diff
@@ -196,7 +196,20 @@
   end
 
   @testset "SpatialPredicatePartitioner" begin
-    # TODO
+    g = RegularGrid{Float64}(10,10)
+    
+    # check if each partition has only 1 point
+    sp = SpatialPredicatePartitioner((x,y) -> norm(x-y) < 1.0)
+    s = subsets(partition(g, sp))
+    @test length(s) == 100
+
+    # check if points x and y belong to the rectangle [0.,5.]x[0.,5.]
+    pred(x, y) = all([0.,0.] .<= x .<=[5.,5.]) && all([0.,0.] .<= y .<= [5.,5.])
+    sp = SpatialPredicatePartitioner(pred)
+    s = subsets(partition(g, sp))
+
+    @test length(s) == 65
```

Comment:
This test is not intuitive at first. Why 65? Can you please design a test with more intuition? Someone reading this later will have a hard time understanding the motivations of the test.